File: gto-support-crashlytics/src/main/java/org/ccci/gto/android/common/crashlytics/timber/CrashlyticsTree.java
package org.ccci.gto.android.common.crashlytics.timber;

import android.util.Log;

import com.crashlytics.android.Crashlytics;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import timber.log.Timber;

public final class CrashlyticsTree extends Timber.Tree {
    private int mLogLevel;
    private int mExceptionLogLevel;

    public CrashlyticsTree() {
        this(Log.INFO, Log.ERROR);
    }

    public CrashlyticsTree(final int logLevel) {
        this(logLevel, logLevel);
    }

    public CrashlyticsTree(final int logLevel, final int exceptionLogLevel) {
        mLogLevel = logLevel;
        mExceptionLogLevel = exceptionLogLevel;
    }

    @Override
    protected boolean isLoggable(@Nullable final String tag, final int priority) {
        return mLogLevel <= priority;
    }

    @Override
    protected void log(final int priority, @Nullable final String tag, @NotNull final String message,
                       @Nullable final Throwable t) {
        Crashlytics.log(priority, tag, message);

        if (t != null && mExceptionLogLevel <= priority) {
            Crashlytics.logException(t);
        }
    }
}

File: gto-support-crashlytics/src/main/java/org/ccci/gto/android/common/crashlytics/timber/CrashlyticsTree.kt
package org.ccci.gto.android.common.crashlytics.timber

import android.util.Log
import com.crashlytics.android.Crashlytics
import timber.log.Timber

@Deprecated(
    "Since v3.6.2, this class uses the deprecated Fabric Crashlytics library that will no longer be supported" +
        " starting November 15th, 2020"
)
class CrashlyticsTree @JvmOverloads constructor(
    private val logLevel: Int,
    private val exceptionLogLevel: Int = logLevel
) : Timber.Tree() {
    constructor() : this(Log.INFO, Log.ERROR)

    override fun isLoggable(tag: String?, priority: Int) = logLevel <= priority
    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        Crashlytics.log(priority, tag, message)
        if (t != null && exceptionLogLevel <= priority) Crashlytics.logException(t)
    }
}
